tests/test_filterLabelFixups.py

```diff
@@ -0,0 +1,118 @@
+        # so both band and physical_filter are implied and not known here.
+        self.calexpMinimalDataId = DataCoordinate.standardize(
+            instrument="HSC", detector=100, visit=903334,
+            universe=self.butler.registry.dimensions,
+        )
+
+    def testReadingOldFileWithIncompleteDataId(self):
+        """If we try to read an old flat with an incomplete data ID, we should
+        get a warning.  It is unspecified what the FilterLabel will have in
+        this case, so we don't check that.
+        """
+        with self.assertWarns(Warning):
+            self.butler.get("flat", self.flatMinimalDataId)
+        with self.assertWarns(Warning):
+            self.butler.get("flat.filterLabel", self.flatMinimalDataId)
+
+    def testFixingReadingOldFile(self):
+        """If we read an old flat with a complete data ID, we fix the
+        FilterLabel.
+        """
+        flatFullDataId = self.butler.registry.expandDataId(self.flatMinimalDataId)
+        flat = self.butler.get("flat", flatFullDataId)
+        self.assertEqual(flat.getFilterLabel().bandLabel, flatFullDataId["band"])
+        self.assertEqual(flat.getFilterLabel().physicalLabel, flatFullDataId["physical_filter"])
+        flatFilterLabel = self.butler.get("flat.filterLabel", flatFullDataId)
+        self.assertEqual(flatFilterLabel.bandLabel, flatFullDataId["band"])
+        self.assertEqual(flatFilterLabel.physicalLabel, flatFullDataId["physical_filter"])
+
+    def testReadingNewFileWithIncompleteDataId(self):
+        """If we try to read a new calexp with an incomplete data ID, the
+        reader should recognize that it can't check the filters and just trust
+        the file.
+        """
+        calexp = self.butler.get("calexp", self.calexpMinimalDataId)
+        calexpFilterLabel = self.butler.get("calexp.filterLabel", self.calexpMinimalDataId)
+        self.assertTrue(calexp.getFilterLabel().hasPhysicalLabel())
+        self.assertTrue(calexp.getFilterLabel().hasBandLabel())
+        self.assertEqual(calexp.getFilterLabel(), calexpFilterLabel)
+
+    def testReadingNewFileWithFullDataId(self):
+        """If we try to read a new calexp with a full data ID, the reader
+        should check the filters in the file for consistency with the data ID
+        (and in this case, find them consistent).
+        """
+        calexpFullDataId = self.butler.registry.expandDataId(self.calexpMinimalDataId)
+        calexp = self.butler.get("calexp", calexpFullDataId)
+        self.assertEqual(calexp.getFilterLabel().bandLabel, calexpFullDataId["band"])
+        self.assertEqual(calexp.getFilterLabel().physicalLabel, calexpFullDataId["physical_filter"])
+        calexpFilterLabel = self.butler.get("calexp.filterLabel", calexpFullDataId)
+        self.assertEqual(calexpFilterLabel.bandLabel, calexpFullDataId["band"])
+        self.assertEqual(calexpFilterLabel.physicalLabel, calexpFullDataId["physical_filter"])
```

Comment:
Worth adding an explicit test that `calexpFullDataId[x]` matches the value read directly from the file (e.g. via `readFits`), as a consistency check of the "find them consistent" statement?

Reply:
:man_shrugging: I think I'd _lean_ towards no; right now, that's a pretty circular check, and I think it's also generally good that these tests strictly test what butler does, and hence isn't too dependent on how that's implemented (even though the _scope_ of the tests are very much tied to a particular piece of that implementation).